src/devices/Mcp3428/Mcp3428.cs

```diff
@@ -0,0 +1,409 @@
+            ResolutionEnum resolution = ResolutionEnum.Bit12, GainEnum pgaGain = GainEnum.X1) : this(i2CDevice)
+        {
+            SetConfig(0, mode: mode, resolution: resolution, pgaGain: pgaGain);
+        }
+
+        /// <summary>
+        /// Protected constructor to easily generate MCP3426/7 devices whose only difference is channel count and I2C address
+        /// </summary>
+        /// <param name="i2CDevice">The i2 c device.</param>
+        /// <param name="channels">The channels.</param>
+        /// <autogeneratedoc />
+        protected Mcp3428(I2cDevice i2CDevice, int channels)
+        {
+            _i2cDevice = i2CDevice;
+            ChannelCount = channels;
+            ReadValue(); // Don't like this in constructor, makes sure props are valid
+        }
+
+        /// <summary>
+        /// Gets the last transmitted bytes. Debug function
+        /// </summary>
+        /// <value>The last bytes.</value>
+        /// <autogeneratedoc />
+        public ReadOnlyMemory<byte> LastBytes => _readBuffer;
+
+        /// <summary>
+        /// Channel most recently read
+        /// </summary>
+        /// <value>The last channel.</value>
+        /// <autogeneratedoc />
+        public byte LastChannel => LastConversion.Channel;
+
+        public ModeEnum Mode
+        {
+            get => _mode;
+            set
+            {
+                WriteConfig(Helpers.SetModeBit(LastConfigByte, value));
+                _mode = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the input gain.
+        /// </summary>
+        /// <value>The pga gain.</value>
+        /// <autogeneratedoc />
+        public GainEnum PGAGain
+        {
+            get => _pgaGain;
+            set
+            {
+                WriteConfig(Helpers.SetGainBits(LastConfigByte, value));
+                _pgaGain = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the bit resolution of the result.
+        /// </summary>
+        /// <value>The resolution.</value>
+        /// <autogeneratedoc />
+        public ResolutionEnum Resolution
+        {
+            get => _resolution;
+            set
+            {
+                WriteConfig(Helpers.SetResolutionBits(LastConfigByte, value));
+                _resolution = value;
+                //WaitTime = (int)(1000.0 / UpdateFrequency(_resolution));
+            }
+        }
+
+        public void Dispose()
+        {
+        }
+
+        /// <summary>
+        /// Reads the channel.
+        /// </summary>
+        /// <param name="channel">The channel.</param>
+        /// <returns>System.Double.</returns>
+        /// <autogeneratedoc />
+        public double ReadChannel(int channel)
+        {
+            return ReadValue(channel);
+        }
+
+        protected const int MaxRetries = 50;
+        protected readonly I2cDevice _i2cDevice;
+        protected readonly byte[] _readBuffer = new byte[3];
```

Comment:
1. why not private?
2. Protected members should start with capital letters (i.e. ReadBuffer)

Reply:
Leftover form previous derived async implementation. Fixed.